ibis/backends/dask/execution/util.py

```diff
@@ -345,3 +348,106 @@ def safe_scalar_type(output_meta):
         output_meta = pd.Timestamp(1, tz=output_meta.tz, unit=output_meta.unit)
 
     return output_meta
+
+
+def add_partitioned_sorted_column(
+    df: Union[dd.DataFrame, dd.Series],
+    set_index: bool = False,
+    col_name: str = "_ibis_index",
+) -> Tuple[dd.DataFrame, List]:
+    """Add a column that is already partitioned and sorted
+    This columns acts as if we had a global index across the distributed data.
+    Important properties:
+    - Each row has a unique id (i.e. a value in this column)
+    - IDs within each partition are already sorted
+    - Any id in partition N_{t} is less than any id in partition N_{t+1}
+    We do this by designating a sufficiently large space of integers per
+    partition via a base and adding the existing index to that base. See
+    `helper` below.
+    Though the space per partition is bounded, real world usage should not
+    hit these bounds. We also do not explicity deal with overflow in the
+    bounds.
+    Parameters
+    ----------
+    df : dd.DataFrame
+        Dataframe to add a column to
+    set_index : bool, optional
```

Comment:
shouldn't we do this explicity? (e.g. in the calling expression)

Reply:
hmm, you need the division, can you ask the df for this?

Reply:
This came out of my previous PR where I used this for grouped udfs (https://github.com/ibis-project/ibis/pull/2786/files#diff-cdacb0d0d05c5f6c1c50b7e78592154aacafc240a769fdf83f51c0945ecae4c0R201). There were cases where it was useful to create this column, but not assign it an index, but also keep around divisions for later (you can only get divisions off the df if you perform the `set_index` space). 

However, now we just `set_index` as part of the start of `selection`. This might simplify all the code here? I can refactor this a bit as part of doing the grouped windows (next on my list).

Reply:
I also feel similar to Jeff that this API looks a bit weird to me that the division of returned separately, and not clear to me when you would want to add an index column but use the division sometime later (and how would you pass the division information to some other node)

Reply:
Is there a case now you actually want not to set the index and just return the division? (or future cases?) If so, it would be helpful to comment on those cases to help readers like myself.